src/crwimage_int.cpp

```diff
@@ -888,12 +888,16 @@ namespace Exiv2 {
         assert(ifdId != ifdIdNotSet);
 
         std::string groupName(Internal::groupName(ifdId));
+        const uint32_t component_size = ciffComponent.size();
+        enforce(component_size % 2 == 0, kerCorruptedMetadata);
+        enforce(component_size/2 <= static_cast<uint32_t>(std::numeric_limits<uint16_t>::max()), kerCorruptedMetadata);
+        const uint16_t num_components = static_cast<uint16_t>(component_size/2);
         uint16_t c = 1;
-        while (uint32_t(c)*2 < ciffComponent.size()) {
+        while (c < num_components) {
```

Comment:
This is the fix for https://github.com/Exiv2/exiv2/security/advisories/GHSA-hqjh-hpv8-8r9p
The value of `ciffComponent.size()` is bounds-checked before the start of the loop and the correct number of iterations is precomputed.